tests/on_target/utils/flash_tools.py

```diff
@@ -33,6 +33,8 @@ def reset_device(serial=SEGGER, reset_kind="RESET_SYSTEM"):
 
 def flash_device(hexfile, serial=SEGGER, extra_args=[]):
     # hexfile (str): Full path to file (hex or zip) to be programmed
+    if hexfile is None:
```

Comment:
I suggest "if not isinstance(hexfile, str)", in case the return for example would be changed to False then this check doesn't work.